### PR TITLE
feat(ci/cd): refactor pipepine (NR-52415)

### PR DESCRIPTION
**Pull request by paologallinaharbur**
The first thing we have to do is to make work the pipeline that is spawned when we do a PR.

It was tested with MongoDB but something in the e2e failed: https://github.com/newrelic/newrelic-prometheus-exporters-packages/actions/runs/3748959020/jobs/6366887256

So I switched to PowerDNS which seem to work better and it worked: https://github.com/newrelic/newrelic-prometheus-exporters-packages/actions/runs/3749070156/jobs/6367129643#step:7:23

So I removed any change of any exporter to not trigger any release: [#973e6a8](https://github.com/newrelic/newrelic-prometheus-exporters-packages/pull/141/commits/973e6a828561ce53a9957e5d74c4d744ccb4f50f)

We still have to refactor the prerelease and release pipelines but I would love to do it in another PR to not make this one even bigger.

### CODE REVIEW

**Comment by paologallinaharbur** [comment]
@kang-makes @sigilioso

**Review by carlossscastro — APPROVED** [review]
👍

**Comment by kang-makes** [comment]
The first thing we have to do is to make work the pipeline that is spawned when we do a PR.

It was tested with mogodb but something in the e2e failed: https://github.com/newrelic/newrelic-prometheus-exporters-packages/actions/runs/3748959020/jobs/6366887256

So I switched to PowerDNS that seem to work better and it worked: https://github.com/newrelic/newrelic-prometheus-exporters-packages/actions/runs/3749070156/jobs/6367129643#step:7:23

So I removed any change of any exporter to not trigger any release: [#973e6a8](https://github.com/newrelic/newrelic-prometheus-exporters-packages/pull/141/commits/973e6a828561ce53a9957e5d74c4d744ccb4f50f)

We still have to refactor the prerelease and release pipelines but I would love to do it in another PR to not make this one even bigger.

File: scripts/build.sh [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+cp "${goreleaser_file_template}" "${goreleaser_file}"
+IFS=',' read -r -a goarchs <<< "$PACKAGE_LINUX_GOARCHS"
+for goarch in "${goarchs[@]}"
+do
+  echo  "Adding ${goarch} to goreleaser"
+  yq e -i ".builds[0].goarch += [ \"${goarch}\" ]" ${goreleaser_file}
+done
+
+GORELEASER_CURRENT_TAG=${VERSION} ${root_dir}/bin/goreleaser build --config ${goreleaser_file} --snapshot --rm-dist
```
Comment (sigilioso) [inline scripts/build.sh:59]:
```suggestion
GORELEASER_CURRENT_TAG=${VERSION} ${root_dir}/bin/goreleaser build --config ${goreleaser_file} --snapshot --rm-dist --id ${goos}
```

**Review by sigilioso — COMMENTED** [review]
The suggestions I left are optional optimizations. We will need to review it again after merging #143 but it think it is almost ready.

File: scripts/build.sh (hunk 1 above)
Comment (kang-makes) [reply scripts/build.sh]: Added also in [#d44cce1](https://github.com/newrelic/newrelic-prometheus-exporters-packages/pull/141/commits/d44cce1e907b131a229265b9e7eb3db3b5725988) for the package step.

**Review by sigilioso — APPROVED** [review]

**Pull request merged**